src/NHibernate.Test/Linq/ByMethod/CastTests.cs

```diff
@@ -29,8 +33,10 @@ where a.Pregnant
 		public void CastDowncast()
 		{
 			var query = session.Query<Mammal>().Cast<Dog>();
+			List<Dog> list;
 			// the list contains at least one Cat then should Throws
-			Assert.That(() => query.ToList(), Throws.Exception);
+			// Do not use bare Throws.Exception due to https://github.com/nunit/nunit/issues/1899
+			Assert.That(() => list = query.ToList(), Throws.InstanceOf<GenericADOException>());
```

Comment:
Two bugs in one. [#47](https://github.com/maca88/AsyncGenerator/issues/47), worked around by adding the local variable, and [#1899](https://github.com/nunit/nunit/issues/1899), worked around by using `InstanceOf`. The later was causing three other tests to fail, they have been "worked around" too.